File: Problems/Algorithms/1768. Merge Strings Alternately/merge_strings.rs
impl Solution {
    pub fn merge_alternately(word1: String, word2: String) -> String {
        let mut chars1 = word1.chars();
        let mut chars2 = word2.chars();
        
        let mut res = vec![];
        for _ in 0..word1.len().max(word2.len()) {
            if let Some(c) = chars1.next() {
                res.push(c);
            }
            
            if let Some(c) = chars2.next() {
                res.push(c);
            }
        }
        
        res.iter().collect()
    }
}

File: Problems/Algorithms/1768. Merge Strings Alternately/merge_strings.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        ans = ""
        n = min(len(word1), len(word2))
        m, l = max(0, len(word1) - n), max(0, len(word2) - n)
        
        for i in range(n):
            ans += word1[i]
            ans += word2[i]
        
        for j in range(m):
            ans += word1[j+n]

        for k in range(l):
            ans += word2[k+n]

        return ans
